src/sentry/api/endpoints/project_group_index.py

```diff
@@ -137,7 +138,10 @@ def _build_query_params_from_request(self, request, project):
 
         query = request.GET.get('query', 'is:unresolved').strip()
         if query:
-            query_kwargs.update(parse_query(project, query, request.user))
+            try:
+                query_kwargs.update(parse_query(project, query, request.user))
+            except DataError:
```

Comment:
Ideally we would simply raise "InvalidQuery" in parse_query, and only catch that here